arch/posix/Kconfig

```
@@ -41,6 +41,18 @@ config ARCH_POSIX_TRAP_ON_FATAL
 	  Raise a SIGTRAP signal on fatal error before exiting.
 	  This automatically suspends the target if a debugger is attached.
 
+config ARCH_POSIX_FORCE_64BIT_TIME_T
+	bool "Set _TIME_BITS=64"
```

Can it be defaulted to `y`?

Maybe there is an obvious answer but are there `native_sim` systems doing CI for Zephyr where we actually want time_t to be 32-bits, in contrast to all other platforms supported by Zephyr? Is this catering to an eventuality? I.e. optimizing for the cold path?

Why not report an error that Zephyr CI should be done with systems that use 64-bit time_t and be done with it?

To be honest, my main concern was the `unit_testing` platform, but it would be good to just be consistent.

```suggestion
	bool
  default y
```

---

The reason is the concern I mentioned here: https://github.com/zephyrproject-rtos/zephyr/issues/90029#issuecomment-3012377380 (_ABI incompatibility with precompiled libraries._)

---

I should rephrase.

Do people actually use C libraries configured with 32-bit `time_t` on 64-bit processors for Zephyr CI?

I suppose, technically, we should ask whether people use 32-bit host processors for Zephyr CI as well.

It just seems like a really marginal use case. Do we even need to support it?

---

> Do people actually use C libraries configured with 32-bit time_t on 64-bit processors for Zephyr CI?

Yes. All Linux distros using glibc in x86-64 have this issue AFAIK, including our CI.

That's why we would provide this option to force it to 64bits. But all we can control is how we build Zephyr and the libraries we build. Whatever came prebuilt from the distro is already built with a 32bit time_t.

---

> As mentioned, the issue with forcing by default glibc to provide a 64bit time_t in 32bit x86 builds, is that other pre-built 32bit libraries in the system had already been built with a 32bit time_t (the glibc default).

Ah, that makes sense. Thanks for clarifying.